src/js/service/CodeOrgMessageService.js

```diff
@@ -0,0 +1,72 @@
+/* Code.org-specific service for managing communication with the outer page
+ * when used as an embedded app in an iframe. */
+(function () {
+  var ns = $.namespace('pskl.service');
+  var MessageType = ExportedConstants.MessageType;
+
+  /**
+   * @param {!ImportService} importService
+   * @constructor
+   */
+  ns.CodeOrgMessageService = function (importService) {
```

Comment:
Having ns.CodeOrgMessageService.prototype.* everywhere feels a little strange to me, though I'm not entirely sure I can justify why. What if you just did:
```
var CodeOrgMessageService = ns.CodeOrgMessageService = function (importService) {
```
and then everywhere else I think you can just do
```
CodeOrgMessageService.prototype.*
```